File: DesafioFullStack.Domain/Entities/Fornecedor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesafioFullStack.Domain.Entities
{
    public class Fornecedor
    {
        public Guid Id { get; set; }
        public string CpfCnpj { get; set; } = string.Empty;
        public string Nome { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Cep { get; set; } = string.Empty;

        // Campos específicos para Pessoa Física
        public string? Rg { get; set; }
        public DateTime? DataNascimento { get; set; }

        public bool EhPessoaFisica => CpfCnpj.Length == 11;
        public DateTime DataCadastro { get; set; }
        public DateTime? DataAtualizacao { get; set; }

        // Relacionamento Many-to-Many
        public ICollection<EmpresaFornecedor> EmpresaFornecedores { get; set; } = new List<EmpresaFornecedor>();
    }
}

File: frontend/src/types/fornecedor.ts
import type { Empresa } from './empresa';

export interface Fornecedor {
    id?: string;
    cpfCnpj: string;
    nome: string;
    email: string;
    cep: string;
    logradouro?: string;
    bairro?: string;
    cidade?: string;
    uf?: string;

    // Campos específicos para Pessoa Física
    rg?: string;
    dataNascimento?: string; 
    
    // Relacionamento
    empresas?: Empresa[];
    createdAt?: string;
    updatedAt?: string;
}

export interface FornecedorFormData {
    cpfCnpj: string;
    nome: string;
    email: string;
    cep: string;
    rg?: string;
    dataNascimento?: string;
    logradouro?: string;
    bairro?: string;
    cidade?: string;
    uf?: string;
}


export const TipoDocumento = {
  CPF: 'CPF',
  CNPJ: 'CNPJ'
} as const;